src/JWT/Builder/JwtBuilder.cs

```diff
@@ -78,6 +78,9 @@ public JwtBuilder AddClaim(string name, object value)
         public JwtBuilder WithSerializer(IJsonSerializer serializer)
         {
             _serializer = serializer;
+#if MODERN_DOTNET
+            SetSerializer(serializer);

```

Comment:
So this one calls the factory by the imported static method? Far from being obvious. Should we call the factory explicitly?

Reply:
Another, bigger problem with this approach is that an instance methods changes the static state. Should it instead instantiate a factory with the given serializer and pass it whenever applicable?